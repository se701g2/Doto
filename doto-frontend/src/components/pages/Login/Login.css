.SettingsPage {
    display: flex;
    flex-direction: row;
}

.google-btn {
    /* width: 184px;
    height: 42px; */
    width: 30vh;
    height: 6vh;
    background-color: #4285f4;
    border-radius: 2px;
    box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.25);
    cursor: pointer;
}
.google-btn .google-icon-wrapper {
    position: absolute;
    /* margin-top: 1px;
    margin-left: 1px; */
    /* width: 40px;
    height: 40px; */
    width: 6vh;
    height: 6vh;
    border-radius: 2px;
    background-color: #fff;
}
.google-btn .google-icon {
    /* margin-top: 11px; */
    /* width: 18px;
    height: 18px; */
    margin-top: 1vh;
    margin-left: 1vh;
    width: 4vh;
    height: 4vh;
}
.google-btn .btn-text {
    padding-top: 1vh;
    margin-left: 7vh;
    color: #fff;
    /* font-size: 14px; */
    font-size: 2.5vh;
    letter-spacing: 0.2px;
}
.google-btn:hover {
    box-shadow: 0 0 6px #4285f4;
}
.google-btn:active {
    background: #1669f2;
}
