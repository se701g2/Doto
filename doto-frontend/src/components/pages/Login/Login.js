import React from "react";
import "./Login.css";

const DotoTitle = () => {
    return (
        <div className="p-4" style={{ marginLeft: "7rem" }}>
            <span className="text-6xl font-bold font"> Doto </span>
        </div>
    );
};

const BlueBubble = () => {
    const url = process.env.BACKEND_URL || "http://localhost:3001";
    return (
        <div className="h-full mr-16" style={{ borderRadius: "0px 79px 0px 0px", backgroundColor: "#3700b3" }}>
            <div className="flex flex-col ml-32 mt-10">
                <div>
                    <span className="text-white font-bold text-5xl">Your life</span>
                </div>
                <div>
                    <span className="text-white text-5xl font-light">Planned for you</span>
                </div>
                <div className="google-btn-container mt-2">
                    <a href={url + "/auth/google"}>
                        <div className="google-btn">
                            <div className="google-icon-wrapper">
                                <img
                                    className="google-icon"
                                    src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                                    alt="signin"
                                />
                            </div>
                            <p className="btn-text">
                                <b>Log in with Google</b>
                            </p>
                        </div>
                    </a>
                </div>
            </div>
        </div>
    );
};

const PurpleBubble = () => {
    return <div className="w-1/3 mt-64" style={{ borderRadius: "79px 0px 0px 0px", backgroundColor: "#8d6cd9" }} />;
};

const Login = () => {
    return (
        <div className="flex flex-row bg-white w-full overflow-hidden max-h-screen">
            <div className="flex flex-col h-screen w-2/3 mt-32">
                <DotoTitle />
                <BlueBubble />
            </div>
            <PurpleBubble />
        </div>
    );
};

export default Login;
