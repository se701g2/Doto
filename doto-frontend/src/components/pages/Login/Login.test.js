import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import Login from "./Login";
import renderer from "react-test-renderer";

describe("<Login /> component being rendered", () => {
    let subject;

    beforeEach(() => {
        subject = mount(<Login />);
    });

    afterEach(() => {
        subject.unmount();
    });

    it("Login component rendered without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Login />, div);
    });

    it("Make sure render matches snapshot", () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
