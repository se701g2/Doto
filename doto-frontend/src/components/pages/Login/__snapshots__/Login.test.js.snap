// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component being rendered Make sure render matches snapshot 1`] = `
<div
  className="flex flex-row bg-white w-full overflow-hidden max-h-screen"
>
  <div
    className="flex flex-col h-screen md:w-2/3 md:mt-32"
  >
    <div
      className="p-4"
      style={
        Object {
          "marginLeft": "7rem",
        }
      }
    >
      <span
        className="text-6xl font-bold font"
      >
         Doto 
      </span>
    </div>
    <div
      className="h-full sm:mr-16 lg:mr-32"
      style={
        Object {
          "backgroundColor": "#3700b3",
          "borderRadius": "0px 79px 0px 0px",
        }
      }
    >
      <div
        className="flex flex-col ml-32 mt-10"
      >
        <div>
          <span
            className="text-white font-bold text-5xl"
          >
            Your life
          </span>
        </div>
        <div>
          <span
            className="text-white text-5xl font-light mr-2"
          >
            Planned for you
          </span>
        </div>
        <div
          className="google-btn-container mt-2"
        >
          <a
            href="https://doto-backend.azurewebsites.net/auth/google"
          >
            <div
              className="google-btn"
            >
              <div
                className="google-icon-wrapper"
              >
                <img
                  alt="signin"
                  className="google-icon"
                  src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                />
              </div>
              <p
                className="btn-text"
              >
                <b>
                  Log in with Google
                </b>
              </p>
            </div>
          </a>
        </div>
        <div
          className="signup-btn-container mt-5"
        >
          <a
            href="https://accounts.google.com/SignUp?continue=https://myaccount.google.com/intro"
          >
            <span
              className="text-white text-2.5vh font-light mr-2"
            >
              Create Google Account
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="md:w-1/3 md:mt-64"
    style={
      Object {
        "backgroundColor": "#8d6cd9",
        "borderRadius": "79px 0px 0px 0px",
      }
    }
  />
</div>
`;
